tests/filecheck/dialects/riscv/riscv_ops.mlir

```diff
@@ -1,267 +1,267 @@
-// RUN: xdsl-opt %s | xdsl-opt --print-op-generic | filecheck %s
+// RUN: xdsl-opt --print-op-generic %s | xdsl-opt | filecheck %s
```

Comment:
Why the print op generic? I think the previous version tested that the custom parser works. It feels like we might want to remove the generic printing entirely from this line.

Reply:
The code coverage should reveal this, or a `print("foo")` in the custom parsing, maybe I'm wrong.

Reply:
The idea was that we test both generic and custom printing, generic printing via the printing and reparsing around the pipe and the custom parsing and printing by parsing and filechecking this file. What command line would you advice @superlopuh?

> The code coverage should reveal this, or a print("foo") in the custom parsing, maybe I'm wrong.

I do not understand this sentence. Could you potentially elaborate?

Reply:
Ah you're right. I'm not sure we need to test that we print and parse custom syntax, though, since that's a well-tested part of the infrastructure already. What's the purpose of the second xdsl-opt? Maybe I'm misunderstanding.

Reply:
We test four things:

- Parsing of custom syntax (via the first xdsl-opt command)
- Printing of generic syntax for this dialect (via the first xdsl-opt command)
- Parsing of generic syntax for this dialect (via the 2nd xdsl-opt command)
- Printing of custom syntax (via the 2nd xdsl-opt command) + filecheck

We can claim that generic parsing is already well tested, but given that we get it at the cost of just a 2nd xdsl-opt call, I feel having it tested per-dialect does not have a large cost.

Reply:
Ok sounds good

Reply:
Upon thinking, I'm not sure the first the `Parsing of custom syntax (via the first xdsl-opt command)` is actually run.

May suggest `xdsl-opt | xdsl-opt --print-op-generic %s | xdsl-opt | filecheck %s`

Reply:
Because the test file may in unknown format yet, (generic format in this case).

Reply:
Indeed, we would need to change the input file to make sure that our custom parser is also tested. I think that would be a good idea, as this will also make the file more readable.

Reply:
But we know that this file is input in generic format? If you wanted to make this purpose explicit you could rename it to `riscv_ops_generic_format.mlir` or something like that? I don't see the appeal of piping it through a bunch of times, what that extra time in testing for every commit for eternity gains in terms of expected bug minimisation.

Reply:
I would rather suggest to turn the input into custom format.

Reply:
Sorry this was just a brain fart by me, I meant custom format...

Reply:
Input file is now in custom format.